docs/architecture-decisions/2024-01-23_init-wizard-v2.md

```diff
@@ -0,0 +1,211 @@
+# AlgoKit Init Wizard Version 2 and Template Enhancements
+
+- **Status:** Proposed (Revision 2)
+- **Owner:** Altynbek Orumbayev
+- **Deciders:** Alessandro (Algorand Foundation), Rob Moore (MakerX), MakerX team
+- **Creation Date:** 2024-01-23
+- **Decision Date:** TBD
+- **Update Date:** 2024-02-12
+
+## Revisions
+
+- **Revision 1:** Initial discussions with Alessandro, Michael, Joe, Georgio, and Chris identified key issues with the existing wizards and explored ways to improve templating capabilities. These discussions led to the consideration of a unified 'monorepo' structure to consolidate template repositories into more focused, smaller generators under a single repository. The revision also sought to refine the wizard and enhance command orchestration capabilities, incorporating tools like `npm workspaces`.
+- **Revision 2:** Engaged in multiple brainstorming sessions with Algorand Foundation/DevRel members and MakerX engineers to develop a pragmatic, time-bound, and prerequisite-neutral strategy. This approach resulted in splitting the ADR into two segments: one focusing on the `init` wizard enhancements and the other on template improvements and command orchestration upgrades in algokit-cli.
+
+## Background
+
+This ADR emerges from various discussions aimed at enhancing the `init` wizard version 2 user experience and aligning it with emerging smart contract languages/frameworks to make it more user-friendly and accessible to beginners. It builds upon prior decisions from the [Advanced algokit generate command](./2023-07-19_advanced_generate_command.md) and [Frontend Templates](./2023-06-06_frontend-templates.md) ADRs, integrating feedback from the Algorand Foundation and DevRel team.
+
+### Main Areas for Improvement
+
+1. **Enhancements to the `Init` Wizard:**
+
+   - Improve user experience by making the wizard more intuitive and less reliant on Algorand-specific jargon.
+   - Streamline the `presets` concept to minimize user inputs and simplify the process.
+
+2. **Template Refinements:**
+
+   - Address potential complexities in maintaining the `fullstack` template with new `smart-contract` template combinations, including future `.NET` integration alongside `puya` compiler-compatible stacks.
+   - Implement `codespaces` configurations for simplified project setup in GitHub Codespaces.
+   - Consider unifying `add smart contract` generators in the `puya` and `beaker` templates into a single, generic generator managed by the CLI itself (suggestion added as part of revision 2 based on DevRel feedback).
```

Comment:
Interested in understanding the problem this solves.

Reply:
@neilcampbell as discussed offline - suggestion certainly will improve aspects around simplifying template building experience and offloading some redundant logic from existing templates but will research a bit deeper and cover this as a follow up once all main pieces within the adr scope are implemented (cc @joe-p )